Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -859,19 +859,28 @@ void MediaPlayerPrivateGStreamerMSE::markEndOfStream(MediaSourcePrivate::EndOfSt
     updateStates();
 }
 
+void MediaPlayerPrivateGStreamerMSE::unmarkEndOfStream()
+{
+    GST_DEBUG("Unmarking end of stream");
+    m_eosPending = false;
+}
+
 MediaTime MediaPlayerPrivateGStreamerMSE::currentMediaTime() const
 {
+    float cachedPosition = m_cachedPosition;
     MediaTime position = MediaPlayerPrivateGStreamer::currentMediaTime();
+    float playbackProgress = std::abs(position.toFloat() - cachedPosition);
 
-    if (m_eosPending && (paused() || (position >= durationMediaTime()))) {
+    if (m_eosPending && (paused() || (std::abs(position.toFloat() - durationMediaTime().toFloat()) <= 1 && playbackProgress == 0))) {
```

Comment:
But looks video decoder reports pts from latest frame, and even pushed buffers for 1s, pts could be reported as 0.75s, (and new frame is waiting to determine new pts), so in this case if there is no playback progress and difference is small, we could say that reached EOS.